### PR TITLE
Debug: Add quotation marks to strings in print statements

### PR DESCRIPTION
**Pull request by ZenBre4ker**
While working on #1222 and #1223 I realized how hard it is to find a problem with valuetypes, that are string.

This should allow to faster see when a value is actually a string and not another valuetype.

If this is useful is still outstanding for discussion.
Helped me to find the problems more easily.

### CODE REVIEW

**Review by TimGoll — APPROVED** [review]
but add an empty line at the file end

File: lua/ttt2/extensions/debug.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+---
+-- Adds quotation marks to strings otherwise just converts them to strings
+-- @param any object The object to convert
+-- @realm shared
+-- @internal
+local function ConvertToString(object)
+	if isstring(object) then
+		return "\"" .. object .. "\""
+	else
```
Comment (Histalek) [inline lua/ttt2/extensions/debug.lua:20]:
```suggestion
	else
```

**Review by Histalek — APPROVED** [review]
Can't hurt having this :)

For the future a `debug.type()` function that can actually distinguish between a `Color` and a `table` might be another possible addition.

**Comment by ZenBre4ker** [comment]
> Can't hurt having this :)
> 
> For the future a `debug.type()` function that can actually distinguish between a `Color` and a `table` might be another possible addition.

Most definitely. Not having to.think about types would be nice, I dislike that you have to convert nil values or use PrintTable for tables. Maybe even allow functions, if possible.
As this will only be used during debugging sessions nobody cares about speed here.

**Pull request merged**